mahta/JS/JS DOM-manipulation/script.js

```diff
@@ -0,0 +1,80 @@
+function generateRandomColor(y, b) {
+  const yellow = Math.floor(Math.random() * (y - 155) + 155);
+  const blue = Math.floor(Math.random() * (b - 100) + 100);
+  return `rgb(${yellow}, ${yellow}, ${blue})`;
+}
+
+const elm = { a: 0, b: 1, c: 2, d: 3, e: 4, f: 5, g: 6, h: 7, i: 8, j: 9 };
+
+function firstState() {
+  document.body.appendChild(elm["a"]);
+  elm["a"].appendChild(elm["b"]);
+  elm["b"].appendChild(elm["c"]);
+  elm["c"].appendChild(elm["d"]);
+  elm["b"].appendChild(elm["e"]);
+  elm["b"].appendChild(elm["g"]);
+  document.body.appendChild(elm["f"]);
+  elm["f"].appendChild(elm["h"]);
+
+  elm["a"].classList.add("container");
+  elm["b"].classList.add("container");
+  elm["c"].classList.add("container");
+  elm["d"].classList.add("container");
+  elm["e"].classList.add("container");
+  elm["f"].classList.add("container");
+  elm["g"].classList.add("container");
+  elm["h"].classList.add("container");
+}
+
+function dif() {
+  elm["b"].removeChild(elm["e"]);
+  elm["b"].removeChild(elm["c"]);
+  elm["g"].appendChild(elm["c"]);
+  elm["f"].appendChild(elm["i"]);
+  elm["h"].appendChild(elm["j"]);
+  elm["i"].classList.add("container");
+  elm["j"].classList.add("container");
+}
+function onButtonClickSet() {
+  document.body.innerHTML = "";
+  const buttonTag = document.createElement("button");
+  buttonTag.textContent = "Reset";
+  buttonTag.addEventListener("click", firstPage);
+  document.body.appendChild(buttonTag);
+  Object.keys(elm).forEach((element) => {
+    elm[element] = document.createElement("div");
+    elm[element].textContent = element;
+  });
+  firstState();
+  dif();
```

Comment:
It is not necessary to create a new node to change an element. Instead, follow the syntax to modify your elements.

```js
 // document.body.innerHTML = "";
  const buttonTag = document.getElementsByTagName("button")[0];
  buttonTag.textContent = "Reset";
  buttonTag.addEventListener("click", firstPage);
  // document.body.appendChild(buttonTag);
  Object.keys(elm).forEach((element) => {
   // elm[element] = document.createElement("div");
    const noode = document.createTextNode(element);
    elm[element].appendChild(noode);
  });
  // firstState();
  dif();

```